lib/phx_component_helpers/css.ex

```diff
@@ -19,9 +19,18 @@ defmodule PhxComponentHelpers.CSS do
 
   defp do_extend_class(assigns_or_options, input_class, default_classes) do
     default_classes =
-      if is_function(default_classes),
-        do: default_classes.(assigns_or_options),
-        else: default_classes
+      cond do
+        is_function(default_classes) ->
+          default_classes.(assigns_or_options)
+
+        is_list(default_classes) ->
+          default_classes
+          |> Enum.filter(&is_binary/1)
+          |> Enum.join(" ")
```

Comment:
I'm wondering if you should also flatten the list?
(I think class lists are flattened in Heex templates)

```suggestion
        is_list(default_classes) ->
          default_classes
          |> List.flatten()
          |> Enum.filter(&is_binary/1)
          |> Enum.join(" ")
```